### PR TITLE
JSON encoder: improve sorting of layout dict keys

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

* Removed `w: 1` from default key entry since it does not get filtered out of the `qmk info` output
* Improved sorting of layout keys: `label`, `x`, `y`, `w`, `h`, `flags` (LED/RGB Matrix), `matrix`
* Make rotary encoder config render on a single line too

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: lib/python/qmk/json_encoders.py [hunk 1]
```diff
@@ -71,22 +80,43 @@ def indent_str(self):
...
+            return '03w'
+
+        elif key == 'h':
+            return '04h'
+
+        elif key == 'flags':
+            return '05flags'
+
+        elif key == 'matrix':
+            return '06matrix'
```
Comment (tzarc) [inline lib/python/qmk/json_encoders.py:116]: How strongly do you prefer this sorting order? Would it be better to put matrix first, given that the firmware-specific stuff actually cares about that, compared to any of the others?

File: lib/python/qmk/json_encoders.py [hunk 2]
```diff
@@ -71,22 +80,43 @@ def indent_str(self):
...
+        elif key == 'x':
+            return '01x'
+
+        elif key == 'y':
+            return '02y'
+
+        elif key == 'w':
+            return '03w'
+
+        elif key == 'h':
```
Comment (tzarc) [inline lib/python/qmk/json_encoders.py:109]: Should we also add 'r' for future-proofing things like XAP or Configurator picking up rotation?

**Review by tzarc — APPROVED** [review]
Just a couple of non-essentials, up to you if you want to address them.

File: lib/python/qmk/json_encoders.py (hunk 1 above)
Comment (noroadsleft) [reply lib/python/qmk/json_encoders.py:116]: The order I've been using for my work is `label`, `matrix`, `x`, `y`, `w`, `h`.

File: lib/python/qmk/json_encoders.py (hunk 2 above)
Comment (fauxpark) [reply lib/python/qmk/json_encoders.py:109]: Not really necessary as `r`,  `rx` and `ry` would be sorted correctly anyway.

File: lib/python/qmk/json_encoders.py (hunk 1 above)
Comment (fauxpark) [inline lib/python/qmk/json_encoders.py:116]:
```suggestion
        elif key == 'matrix':
            return '01matrix'

        elif key == 'x':
            return '02x'

        elif key == 'y':
            return '03y'

        elif key == 'w':
            return '04w'

        elif key == 'h':
            return '05h'

        elif key == 'flags':
            return '06flags'
```

File: lib/python/qmk/tests/test_cli_commands.py [hunk 3]
```diff
@@ -291,7 +291,7 @@ def test_generate_version_h():
 def test_format_json_keyboard():
     result = check_subcommand('format-json', '--format', 'keyboard', 'lib/python/qmk/tests/minimal_info.json')
     check_returncode(result)
-    assert result.stdout == '{\n    "keyboard_name": "tester",\n    "maintainer": "qmk",\n    "layouts": {\n        "LAYOUT": {\n            "layout": [\n                { "label": "KC_A", "matrix": [0, 0], "x": 0, "y": 0 }\n            ]\n        }\n    }\n}\n'
+    assert result.stdout == '{\n    "keyboard_name": "tester",\n    "maintainer": "qmk",\n    "layouts": {\n        "LAYOUT": {\n            "layout": [\n                {"label": "KC_A", "x": 0, "y": 0, "matrix": [0, 0]}\n            ]\n        }\n    }\n}\n'
```
Comment (fauxpark) [inline lib/python/qmk/tests/test_cli_commands.py:294]:
```suggestion
    assert result.stdout == '{\n    "keyboard_name": "tester",\n    "maintainer": "qmk",\n    "layouts": {\n        "LAYOUT": {\n            "layout": [\n                {"label": "KC_A", "matrix": [0, 0], "x": 0, "y": 0}\n            ]\n        }\n    }\n}\n'
```

File: lib/python/qmk/tests/test_cli_commands.py [hunk 4]
```diff
@@ -303,7 +303,7 @@ def test_format_json_keymap():
 def test_format_json_keyboard_auto():
     result = check_subcommand('format-json', '--format', 'auto', 'lib/python/qmk/tests/minimal_info.json')
     check_returncode(result)
-    assert result.stdout == '{\n    "keyboard_name": "tester",\n    "maintainer": "qmk",\n    "layouts": {\n        "LAYOUT": {\n            "layout": [\n                { "label": "KC_A", "matrix": [0, 0], "x": 0, "y": 0 }\n            ]\n        }\n    }\n}\n'
+    assert result.stdout == '{\n    "keyboard_name": "tester",\n    "maintainer": "qmk",\n    "layouts": {\n        "LAYOUT": {\n            "layout": [\n                {"label": "KC_A", "x": 0, "y": 0, "matrix": [0, 0]}\n            ]\n        }\n    }\n}\n'
```
Comment (fauxpark) [inline lib/python/qmk/tests/test_cli_commands.py:306]:
```suggestion
    assert result.stdout == '{\n    "keyboard_name": "tester",\n    "maintainer": "qmk",\n    "layouts": {\n        "LAYOUT": {\n            "layout": [\n                {"label": "KC_A", "matrix": [0, 0], "x": 0, "y": 0}\n            ]\n        }\n    }\n}\n'
```

**Pull request merged**